### ISSUE TITLE
[ComboBox]: (React component) List box is still in DOM when "hidden" visually

### ISSUE THREAD

**Issue by nertzy** [carbon-design-system/carbon#5949]
## Detailed description

> What version of the Carbon Design System are you using?

carbon-components-react@7.11.3

> What did you expect to happen?

When the ComboBox's drop down is open, I expect to see the element as normal.

When it is "closed", I expect the element to either have the `hidden` attribute, the `aria-hidden` attribute, or not be present.

> What happened instead?

The element is always present and never marked as hidden. When it is visually "hidden", it has a CSS max-height of zero and therefore doesn't render to seeing users, but is still present otherwise.

## Steps to reproduce the issue

1. Render the ComboBox component
2. Notice that the `[role=listbox]` element is present in the DOM, when it shouldn't be.
3. Open the ComboBox
4. Notice that the `[role=listbox]` element is still present in the DOM, which is good.
5. Close the ComboBox
6. Notice that the `[role=listbox]` element is still present in the DOM, when it shouldn't be.

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/intelligent-meitner-68xzq?file=/src/index.js

**Comment by joshblack** [comment]
cc @dakahn is this handled by your downshift updates? 👀

**Comment by dakahn** [comment]
@joshblack sure is! 

closed by https://github.com/carbon-design-system/carbon/pull/5373
